### PR TITLE
New transformations and correlation algorithms

### PR DESCRIPTION
**Pull request by ErichZimmer**
New:
+ `rotate_image` transformation 
+ `skew_image` transformation
+ `fft_norm_correlate_images` for dedicated normalized correlation (uses np.int16 dtype)
+ `phase_correlation` with option for normalized correlation (uses np.int16 dtype)

Changes:
+ `find_all_second_peaks` doesn't use numpy masked array for increased speed
+ `vectorized_correlation_to_displacements` has offset correlation mode to eliminate negative correlation issue for gaussian method
+ sigma <= 0 disables lowpass filter in `gen_lowpass_background` 
+ fixed minor bug and increased stability in `local_variance_normalization`
+ added interpolation order for `stretch_image`

### CODE REVIEW

File: openpiv/pyprocess.py [hunk 1]
```diff
@@ -340,14 +344,18 @@ def find_all_second_peaks(corr, width = 2):
...
     jini[jini < 0] = 0
     jfin[jfin > corr.shape[2]] = corr.shape[2]
-    # create a masked view of the corr
-    tmp = corr.view(np.ma.MaskedArray)
+    # create a masked view of the corr, is too slow?
+    #tmp = corr.view(np.ma.MaskedArray)
+    tmp = corr.copy()
     for i in ind:
-        tmp[i, iini[i]:ifin[i], jini[i]:jfin[i]] = np.ma.masked
+        tmp[i, iini[i]:ifin[i], jini[i]:jfin[i]] = 0 #ma.masked
```
Comment (alexlib) [inline openpiv/pyprocess.py:351]: i'm not sure but I think 0 might cause issues with non-clipped correlation maps. we had in the past some negative correlations and i think it's a reason to use np.ma.masked

Comment (ErichZimmer) [reply openpiv/pyprocess.py:351]: I'll check other solutions, but  not using numpy masked arrays were up to 3x faster for large sliding window arrays. I do see a severe issue with negative correlation values, so I'll go ahead to do more testing for a better solution before we go back to masked arrays.

Comment (alexlib) [reply openpiv/pyprocess.py:351]: maybe we use np.ma.masked incorrectly. I think we shall try to make `tmp = np.ma.array()` first and then just add mask whenever it's needed. 
how do you test for memory use? I'll try this solution

Comment (ErichZimmer) [reply openpiv/pyprocess.py:351]: For memory usage on numpy arrays, I use np.nbytes() and divide by 1e6 to get MB.

**Comment by ErichZimmer** [comment]
How should the pre-processing filters be tested? They pass visual inspections and comparisons with commercial softwares, but I can't quite rap my mind around testing pre-processing filters. All other functions are now tested and pass.

**Comment by alexlib** [comment]
I think we need to creat e a small synthetic test case that fails and another that passes the filter. Just to know that it works as specified. Other tests and notebooks will be visually compared to see that major things do not  change

**Comment by ErichZimmer** [comment]
On a side note, a vector validation function I made based off the z-score shows some promising results. It operates similar to the global STD validation filter, so I don't see too much of an advantage yet.

**Comment by alexlib** [comment]
@ErichZimmer - please update on the status of the tests - shall we merge or wait or maybe drop this pull request?

**Comment by alexlib** [comment]
great. waiting. no hurry.

**Comment by ErichZimmer** [comment]
I'm going to make a separate pull request for the minor bug fixes and slight performance increases with the peak searching algorithms.

**Comment by alexlib** [comment]
@ErichZimmer - do we have enough tests for this pull request?

**Comment by ErichZimmer** [comment]
Not yet. I'm doing some more synthetic tests, but things are going slowly because of work and the construction of a new tornado-like vortex simulator for PIV experiments. Furthermore, I'm getting values greater than one for the normalized correlation, so something might be wrong.

**Comment by ErichZimmer** [comment]
The synthetic tests will take a while as there are over 10,000 image pairs testing particle diameter, particle density, particle velocity, effects of noise and pair loss, effects of shear and Rankine vortices, and effects of uneven illumination.

**Comment by ErichZimmer** [comment]
I'm going to make a new pull request and close this one since my branch now only has enhancements to already existing functions (there will be another pull request for new functions and algorithms).

**Pull request closed without merging**